File: stellar-insured-contracts/sdk/mobile/common/contract_interface.dart
# Mobile-Optimized Contract Interface (Dart)

abstract class PropChainMobileSDK {
  Future<PropertyInfo> getPropertyInfo(String propertyId);
  Future<TransactionResult> transferProperty(String propertyId, String to);
  Future<String> signTransactionOffline(Map<String, dynamic> txData);
  Future<String> scanQRCode();
  void subscribeToPropertyUpdates(String propertyId, void Function(dynamic update) callback);
  Future<bool> authenticateBiometric();
  void onError(void Function(String error) callback);
}

class PropertyInfo {
  final String id;
  final String owner;
  final String location;
  final double valuation;
  final String metadataUri;

  PropertyInfo(this.id, this.owner, this.location, this.valuation, this.metadataUri);
}

class TransactionResult {
  final String txHash;
  final String status; // 'pending', 'confirmed', 'failed'
  final String? error;

  TransactionResult(this.txHash, this.status, [this.error]);
}

File: stellar-insured-contracts/sdk/mobile/common/contract_interface.ts
// Mobile-Optimized Contract Interface (TypeScript)
// This file defines the main interface for interacting with PropChain contracts from mobile apps.

export interface PropertyInfo {
  id: string;
  owner: string;
  location: string;
  valuation: number;
  metadataUri: string;
}

export interface TransactionResult {
  txHash: string;
  status: 'pending' | 'confirmed' | 'failed';
  error?: string;
}

export interface PropChainMobileSDK {
  getPropertyInfo(propertyId: string): Promise<PropertyInfo>;
  transferProperty(propertyId: string, to: string): Promise<TransactionResult>;
  signTransactionOffline(txData: object): Promise<string>; // Returns signed tx
  scanQRCode(): Promise<string>; // Returns scanned data
  subscribeToPropertyUpdates(propertyId: string, callback: (update: any) => void): void;
  authenticateBiometric(): Promise<boolean>;
  onError(callback: (error: string) => void): void;
}
